base/util/slice/slice.go

```diff
@@ -71,11 +71,11 @@ func StringSliceMerge(slices ...[]string) []string {
 	return result
 }
 
-func ParseCommaSeparatedList(list string) set.Set[string] {
+func ParseCommaSeparatedList(list string) []string {
```

Comment:
This would confuse me. If we return a set I know that the order is not preserved. When returning a slice I would expect the order is the same as in the original string. This not not the case because we are internally still using a set. I would or (1) revert it to return a set, or (2) refactor the method so the order is preserved.